### PR TITLE
Yul codegen for break & continue statements

### PR DESCRIPTION
**Pull request by christianparpart**
**It would make sense to merge #6315 first**

Closes https://github.com/ethereum/solidity/issues/4576

It contains the codegen part of Yul's break & continue statements.

### Checklist
* [x] implement codegen
* [x] Tests (codeegeneration tested via semantic tests)
* [x] ChangeLog entry

### CODE REVIEW

**Comment by chriseth** [comment]
Changelog? But can also be part of the next PR.

**Review by chriseth — DISMISSED** [review]

**Comment by chriseth** [comment]
Probably the best way to test this is using solidity inline assembly.

**Comment by leonardoalt** [comment]
Needs rebase. Does this PR depend on any other one?

**Comment by christianparpart** [comment]
@leonardoalt rebased. it kinda depends on #6315 to be merged first. which is in `Waiting-for-Review` mode :)

**Comment by chriseth** [comment]
Needs rebase again.

**Comment by axic** [comment]
> It would make sense to merge #6315 first

Marked it WIP because of this.

**Comment by chriseth** [comment]
Rebasing on top of #6315 mainly to enable fuzzing both.

**Comment by chriseth** [comment]
That's the file that causes the failure and it looks really bad: https://github.com/dapphub/dappsys-monolithic/blob/master/note.sol

It is a logging modifier that just records `callvalue`...

**Comment by chriseth** [comment]
Changed PR base.

File: .circleci/config.yml [hunk 1]
```diff
@@ -421,3 +421,17 @@ workflows:
           <<: *build_on_tags
           requires:
             - build_x86_archlinux
+  test_nightly:
```
Comment (chriseth) [inline .circleci/config.yml]: Hm, why did that end up here?

Comment (chriseth) [reply .circleci/config.yml]: Ah, the PR base was wrong. Fixed.

Comment (leonardoalt) [reply .circleci/config.yml]: There are some commits from @erak , not sure how that happened

Comment (christianparpart) [reply .circleci/config.yml]: probably the interactive rebase slipped that in? (maybe check `git log ...` for unexpected commits?)

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**